File: leetcode2/0326. Power of Three.py
class Solution(object):
    def isPowerOfThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n<=0:
            return False
        
        rep=3**10
        while rep<2**31-1:
            rep=rep*3
        rep=rep//3
        
        return rep%n==0

File: archives/C#/0326. Power of Three.cs
public class Solution {
    public bool IsPowerOfThree(int n) {
        if (n<=0){
            return false;
        }
        int rep=3;
        while(rep<(int.MaxValue/3)){
            rep*=3;
        }
        return rep%n==0;
    }
}
